src/components/utils/TextInputWrapper/TextInputWrapper.style.ts

```diff
@@ -0,0 +1,173 @@
+import { css, SerializedStyles } from '@emotion/core';
+import { darken, lighten, rem } from 'polished';
+import { Props } from './TextInputWrapper';
+import { Theme } from 'theme';
+import { DEFAULT_SIZE, getTextFieldSize } from 'utils/size-utils';
+
+const wrapperStyleSwitch = (
+  theme: Theme,
+  lean?: boolean,
+  error?: boolean,
+  styleType?: string,
+  disabled?: boolean
+) => {
+  if (lean) {
+    return `
+    `;
+  }
+
+  switch (styleType) {
+    case 'outlined':
+      return `
+        box-shadow: 0 0 0 1px
+          ${theme.utils.getColor('lightGray', 400)};
+        &:focus-within, &:hover {
+          box-shadow: 0 0 0 1px ${
+            !disabled ? 'transparent' : theme.utils.getColor('lightGray', 400)
+          };
+        }
+      `;
+    case 'elevated':
+      return `
+        box-shadow: ${disabled ? 'initial' : theme.elevation['01']};
+      `;
+    case 'filled':
+    default:
+      return ``;
+  }
+};
+/**
+ * this wrapper must remain simple and not mess with children properties as it will be used
+ * in custom implementation needed eg: datepicker
+ * */
+export const wrapperStyle = ({ disabled, locked, status, lean, styleType, dark }: Props) => (
+  theme: Theme
+): SerializedStyles => {
+  const error = status === 'error';
+  const backgroundColor = dark ? theme.utils.getColor('darkGray', 600) : theme.palette.white;
+
+  return css`
+    transition: background-color 0.25s, box-shadow 0.25s, border-color 0.25s;
+    border-radius: ${theme.spacing.xsm};
+    cursor: ${disabled || locked ? 'not-allowed' : 'auto'};
+    flex: 1 1 100%;
+    user-select: none;
+    position: relative;
+    background-color: ${backgroundColor};
+    opacity: ${disabled && 0.5};
+    border: 2px solid transparent;
+    ${wrapperStyleSwitch(theme, lean, error, styleType, Boolean(disabled || locked))}
+    border-color: ${error ? theme.utils.getColor('error', 400, 'normal') : undefined};
+
+    ${!disabled &&
+      !locked &&
+      `&:hover {
+      background-color: ${dark ? lighten(0.1, backgroundColor) : darken(0.03, backgroundColor)};
+      border-color: ${styleType === 'outlined' && !error && theme.utils.getColor('lightGray', 400)};
+      box-shadow: ${styleType === 'elevated' && theme.elevation['02']};
+    }
+    `}
+
+  &:focus-within {
+      border-color: ${!error && theme.utils.getColor('lightGray', 400)};
+      box-shadow: ${styleType === 'elevated' && theme.elevation['02']};
+      background-color: ${theme.palette.white};
+    }
+
+    ${disabled ||
+      (locked &&
+        `
+      &:before {
+        content: '';
+        background-color: rgba(255, 255, 255, 0.15);
+        height: 100%;
+        width: 100%;
+        position: absolute;
+        left: 0;
+        top: 0;
+        z-index: 1;
+      }
+      * {
```

Comment:
These rules may create UI inconsistencies in the future. We should refactor CSS to avoid using `*` (select all operators) and `!important` tags.

Reply:
because this case is in disabled or locked state this should replace all. Its way easier and readable in my opinion this way rather than having 100 if elses here and there. What do you think?

Reply:
After a close look at this, I think the logic operator here should change to `(disabled || locked) && ...` the parenthesis will change the behavior of the styles.

Since we manage the colors and not the users, we can remove `!important` tags and create rules for specific elements.

example:
```
     > input,textarea {
        color: ${theme.utils.getColor('lightGray', 600)};
        fill: ${theme.utils.getColor('lightGray', 600)};
      }
```